### PR TITLE
FileManager: add code snippets to the Getting Started

### PR DESCRIPTION
**Pull request by AnnaKondratova**

### CODE REVIEW

File: concepts/05 Widgets/FileManager/00 Getting Started with File Manager/10 Create a File Manager.md [hunk 1]
```diff
@@ -63,4 +63,224 @@ The following code creates the **FileManager** widget and adds it to your page.
         // ...
     ];      
 
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-file-manager id="fileManager"
+        [fileSystemProvider]="fileItems">
+        <dxo-permissions 
+            create="true"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/00 Getting Started with File Manager/10 Create a File Manager.md]: The same change applies to all Boolean attributes.

```suggestion
            [create]="true"
```

File: concepts/05 Widgets/FileManager/00 Getting Started with File Manager/10 Create a File Manager.md [hunk 2]
```diff
@@ -63,4 +63,224 @@ The following code creates the **FileManager** widget and adds it to your page.
...
+    import { Component } from '@angular/core';    
+
+    @Component({
+        selector: 'app-root',
+        templateUrl: './app.component.html',
+        styleUrls: ['./app.component.css']
+    })
+
+    export class AppComponent {
+        fileItems:[{
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/00 Getting Started with File Manager/10 Create a File Manager.md]:
```suggestion
        fileItems = [{
```

File: concepts/05 Widgets/FileManager/00 Getting Started with File Manager/10 Create a File Manager.md [hunk 3]
```diff
@@ -63,4 +63,224 @@ The following code creates the **FileManager** widget and adds it to your page.
...
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-file-manager id="fileManager"
+        [fileSystemProvider]="fileItems">
+        <dxo-permissions 
+            create="true"
+            copy="true"
+            move="true"
+            remove="true"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/00 Getting Started with File Manager/10 Create a File Manager.md]: I think I saw this option being renamed to `delete`.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — APPROVED** [review]

File: concepts/05 Widgets/FileManager/00 Getting Started with File Manager/10 Create a File Manager.md [hunk 4]
```diff
@@ -63,4 +63,224 @@ The following code creates the **FileManager** widget and adds it to your page.
...
+        templateUrl: './app.component.html',
+        styleUrls: ['./app.component.css']
+    })
+
+    export class AppComponent {
+        fileItems = [{
+            name: "Documents",
+            isDirectory: true,
+            items: [{
+                    name: "Projects",
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/00 Getting Started with File Manager/10 Create a File Manager.md]: Please fix indentation across all these code examples. General rule: each next nesting level should be indented by 4 spaces from the previous level.

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md [hunk 5]
```diff
@@ -78,6 +78,194 @@ In the example below, the **FileManager** widget displays hierarchical data stor
     //...
     ];
 
+##### Angular
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:63]: Does this example differ from the one we give in the "Bind the File Manager Widget to a File System" article?

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md [hunk 6]
```diff
@@ -138,6 +327,216 @@ If the data source's field names differ from the standard field names mentioned
...
+
+    @Component({
+        selector: 'app-root',
+        templateUrl: './app.component.html',
+        styleUrls: ['./app.component.css']
+    })
+
+    export class AppComponent {
+        fileItems: object[];
+        objectFileProvider: ObjectFileSystemProvider;
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:313]: This type should be imported from somewhere.

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md [hunk 7]
```diff
@@ -138,6 +327,216 @@ If the data source's field names differ from the standard field names mentioned
...
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';     
+
+    import { DxFileManager } from 'devextreme-vue/file-manager';
+    import ObjectFileSystemProvider from 'devextreme/file_management/object_provider';
+    import { fileItems } from './data.js';
+
+    const objectFileProvider = new ObjectFileSystemProvider({
+        data:fileItems,
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md]:
```suggestion
        data: fileItems,
```

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md [hunk 8]
```diff
@@ -138,6 +327,216 @@ If the data source's field names differ from the standard field names mentioned
...
+        templateUrl: './app.component.html',
+        styleUrls: ['./app.component.css']
+    })
+
+    export class AppComponent {
+        fileItems: object[];
+        objectFileProvider: ObjectFileSystemProvider;
+
+        constructor() {
+            this.fileItems = [{
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:316]:
```suggestion
            const fileItems = [{
```

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md [hunk 9]
```diff
@@ -138,6 +327,216 @@ If the data source's field names differ from the standard field names mentioned
...
+                        itemSize: 1024
+                    },{
+                        name: "Passwords.rtf",
+                        isFolder: false,
+                        itemSize: 2048
+                    }]
+                }]
+            }];
+            this.objectFileProvider = new ObjectFileSystemProvider({
+                data: this.fileItems,
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:334]:
```suggestion
                data: fileItems,
```

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/30 Custom File System.md [hunk 10]
```diff
@@ -2,59 +2,205 @@ Use the **FileManager** widget's [Custom](/api-reference/10%20UI%20Widgets/dxFil
...
+Follow the steps below to implement custom file system provider in the widget.
 
----
+1. Assign the [Custom](/api-reference/10%20UI%20Widgets/dxFileManager/5%20File%20System%20Providers/Custom '/Documentation/ApiReference/UI_Widgets/dxFileManager/File_System_Providers/Custom/') file system provider to the [fileSystemProvider](/api-reference/10%20UI%20Widgets/dxFileManager/1%20Configuration/fileSystemProvider.md '/Documentation/ApiReference/UI_Widgets/dxFileManager/Configuration/#fileSystemProvider') option to implement a custom file system provider and bind the **FileManager** widget to it.
 
-##### jQuery
+2. Use the [getItems](/api-reference/10%20UI%20Widgets/dxFileManager/5%20File%20System%20Providers/Custom/1%20Configuration/getItems.md '/Documentation/ApiReference/UI_Widgets/dxFileManager/File_System_Providers/Custom/Configuration/#getItems') option to get file system items. The **[fieldName]Expr** options specify the attribute names that store file system item keys, names, sizes, modification dates, and etc.
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/10 Bind to File Systems/30 Custom File System.md]: `getItems` option? It is named as if it were a method.

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/30 Custom File System.md [hunk 11]
```diff
@@ -2,59 +2,205 @@ Use the **FileManager** widget's [Custom](/api-reference/10%20UI%20Widgets/dxFil
...
+                deleteItem                
+            });
+        }
+        function getItems(pathInfo) {
+            // ...
+        }
+        function createDirectory(parentDirectory, name) {
+            // ...
+        } 
+        function deleteItem(item) {
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/FileManager/10 Bind to File Systems/30 Custom File System.md]: I'm not sure that this code is valid. Functions cannot be declared straight in the class.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md (hunk 5 above)
Comment (AnnaKondratova) [reply concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:63]: Yes, with a small difference: in the current article the example illustrates binding to a data source with non conventional field names.
In the 'Getting Started' section we intentionally use the Object file provider as an example but in its simple version. See a note in the article (Getting Started).

Comment (AnnaKondratova) [reply concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:63]: The difference in these articles is correct.

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md (hunk 8 above)
Comment (AnnaKondratova) [reply concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:316]: The fileItems is declared above, so it correct to use it in such way
export class AppComponent {
        fileItems: object[];
        ...
        constructor() {
            this.fileItems = ... 
}

Explain why you change this?

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md (hunk 6 above)
Comment (RomanTsukanov) [reply concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:313]: I still see no import

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md (hunk 8 above)
Comment (RomanTsukanov) [reply concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:316]: I didn't notice the declaration in the class. Ignore the suggestion.

File: concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md (hunk 5 above)
Comment (AnnaKondratova) [reply concepts/05 Widgets/FileManager/10 Bind to File Systems/10 Object File System.md:63]: I decided to leave two examples (simple and advanced binding) in the 'Object File System' topic as if a user enters documentation in Binding section (passing the Getting Started section), the user will see only one way of binding.

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**